com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs

```diff
@@ -684,13 +687,13 @@ public override void Setup(ScriptableRenderContext context, ref RenderingData re
 #endif
 
                 // handle multisample depth resolve by setting the appropriate store actions if supported
-                if (requiresDepthCopyPass && cameraTargetDescriptor.msaaSamples > 1 && RenderingUtils.MultisampleDepthResolveSupported(useRenderPassEnabled))
+                if ((requiresDepthCopyPass || postProcessingUseDepthTextureMainPass) && cameraTargetDescriptor.msaaSamples > 1 && RenderingUtils.MultisampleDepthResolveSupported(useRenderPassEnabled))
                 {
                     bool isCopyDepthAfterTransparent = m_CopyDepthPass.renderPassEvent == RenderPassEvent.AfterRenderingTransparents;
 
                     // we could StoreAndResolve when the depth copy is after opaque, but performance wise doing StoreAndResolve of depth targets is more expensive than a simple Store + following depth copy pass on Apple GPUs,
                     // because of the extra resolve step. So, unless we are copying the depth after the transparent pass, just Store the depth target.
-                    if (isCopyDepthAfterTransparent && !copyColorPass)
+                    if ((isCopyDepthAfterTransparent || postProcessingUseDepthTextureMainPass) && !copyColorPass)
```

Comment:
Nitpicking: you have in multiple places isCopyDEpthAfterTransparent || postProcessingUseDepthTextureMainPass maybe you could give have a variable with some meaning for this to improve readability. The logic of store actions is a little bit convoluted, but I guess the plan is to improve it with RenderGraph anyway, so no big deal.